### ISSUE TITLE
Continued issues with ProxyObject

### ISSUE THREAD

**Issue by wagenet** [typed-ember/glint#126]
The `{{get}}` helper doesn't error, but seems to fail to infer the type.

```
app/components/deploy-list.hbs:31:46 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'string | number | boolean | void | SafeString | AcceptsBlocks<{}> | null'.
  Type 'unknown' is not assignable to type 'AcceptsBlocks<{}>'.

31                     <a class="sha-link" href={{commitUrl}}>
```

```hbs
                {{#let (get deploy.githubCommit 'url') as |commitUrl|}}
                  {{#if commitUrl}}
                    <a class="sha-link" href={{commitUrl}}>
                      {{deploy.shortSha}}
                    </a>
                  {{else}}
                    {{deploy.shortSha}}
                  {{/if}}
                {{/let}}
```

```ts
export default class GithubCommit extends Model {
  @attr('string')
  declare url: string | null;

  // ...
}
```

**Comment by wagenet** [comment]
Looks like the issue is that `deploy.githubCommit` is `GithubCommit | undefined`.

**Comment by wagenet** [comment]
May be fixed by #127

**Comment by wagenet** [comment]
I believe so.
